### PR TITLE
USE ONE SET OF METADATA MULTIPLE TIMES

### PR DESCRIPTION
**Pull request by theomarkkuspaul**
This change addresses the need by:
* comment relevant js logic
* add `hidden` class to dam dashboard stylesheets
* add wrapping form to dam dashboard uploader to house hidden image metadata input
* add file upload counter partial
* implement feature to keep count of uploaded image count
* develop logic to track which dam media files have been uploaded
* implement submit button which posts ids of successfully uploaded dam media

### CODE REVIEW

File: app/controllers/concerns/dam_authentication.rb [hunk 1]
```diff
@@ -13,4 +13,8 @@ def user_not_authorized
   def pundit_user
       current_dam_user
   end
+
+  def authorize_collection(array)
```
Comment (EdwinRozario) [inline app/controllers/concerns/dam_authentication.rb]: call it items not array

Comment (EdwinRozario) [reply app/controllers/concerns/dam_authentication.rb]: always do like `books.each do |book|`

File: app/controllers/dam_bulk_uploader_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+class DamBulkUploaderController < ApplicationController
+  include DamAuthentication
+
+  def new
+    @dam_resources = dam_media_params
```
Comment (EdwinRozario) [inline app/controllers/dam_bulk_uploader_controller.rb]: This is a Java way of writing code. Please do

`@dam_resources = dam_media_params.split(',').map do |id|
                                     DamResource.find(id)
                                   end`

File: app/controllers/dam_bulk_uploader_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    .split(',')
+    .map do |id|
+      DamResource.find(id)
+    end
+
+    authorize_collection @dam_resources
+  end
+
+  def create
+    ids = dam_resources_params.map { |dr| dr.first }
```
Comment (EdwinRozario) [inline app/controllers/dam_bulk_uploader_controller.rb]: You are still not using Rubocop aye? Not `dr` variables should be meaningful.

`dam_resources_params.map { |resource| resource.first }` 
or
`dam_resources_params.map { |param| param.first }`

File: app/services/metadata.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+
+class Metadata
+
```
Comment (EdwinRozario) [inline app/services/metadata.rb]: extra new line

File: app/services/metadata.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  attr_reader :params, :dam_resource_id
+
+  def initialize(params, dam_resource_id)
+    @params = params.as_json.symbolize_keys
+    @dam_resource_id = dam_resource_id
+  end
+
+  def generate_dam_metadata
+    metadata = params
```
Comment (EdwinRozario) [inline app/services/metadata.rb]:
```Ruby
metadata = params.slice(:description, :restrictions, 
                        :copyright_owner, :date, :expiry_date,
                        :dam_classifications, :dam_people,
                        :is_consent, :consent_file,
                        :dam_locations).merge(dam_resources_params)
```

File: app/services/metadata.rb (hunk 4 above)
Comment (theomarkkuspaul) [reply app/services/metadata.rb]: 👍

File: app/controllers/concerns/dam_authentication.rb (hunk 1 above)
Comment (theomarkkuspaul) [reply app/controllers/concerns/dam_authentication.rb]: 👍

File: app/controllers/dam_bulk_uploader_controller.rb (hunk 2 above)
Comment (theomarkkuspaul) [reply app/controllers/dam_bulk_uploader_controller.rb]: 👍

File: app/controllers/dam_bulk_uploader_controller.rb (hunk 3 above)
Comment (theomarkkuspaul) [reply app/controllers/dam_bulk_uploader_controller.rb]: 👍

**Pull request merged**